src/platform/magento2/o2m.js

```diff
@@ -27,301 +12,41 @@ function isNumeric (val) {
- * @param {Object} config global CLI configuration
+ * @param  job
  * @param {Function} done callback - @example done(new Error()) - to acknowledge problems
  */
-function processSingleOrder (orderData, config, job, done, logger = console) {
-  const TOTAL_STEPS = 4;
+async function processSingleOrder (orderData, job, done, logger = console) {
   const THREAD_ID = 'ORD:' + (job ? job.id : 1) + ' - '; // job id
-  let currentStep = 1;
-
-  /**
-   * Internal function to compose Error object using messages about other errors.
-   *
-   * 'Error' constructor should contain one message object only.
-   * (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error)
-   *
-   * @param {string} message Main error message.
-   * @param {string|array|object} errors Additional error message or error object or array of array objects.
-   * @return {Error}
-   */
-  function composeError (message, errors) {
-    if (typeof errors === 'string') {
-      message = message + ' ' + errors;
-    } else if (Array.isArray(errors)) {
-      // case with array of validation errors (ajv.ErrorObject - node_modules/ajv/lib/ajv.d.ts)
-      errors.forEach((item) => {
-        const part = (typeof item === 'string') ? item : (item.message || '');
-        message = (message + ' ' + part).trim();
-      });
-    } else if (errors && (errors.message || errors.errorMessage)) {
-      // I don't know possible structure of an 'errors' in this case, so I take 'apiError()' from 'src/lib/util.js'
-      // we should use debugger to inspect this case in more details and modify code.
-      message = message + ' ' + (errors.message || errors.errorMessage);
-    }
-    return new Error(message.trim());
-  }
 
-  if (!validate(orderData)) { // schema validation of upcoming order
-    logger.error(THREAD_ID + ' Order validation error!', validate.errors);
-    done(composeError('Error while validating order object.', validate.errors));
-
-    if (job) job.progress(currentStep++, TOTAL_STEPS);
+  if (!orderProcessor.validateOrder({ orderData, job, done })) {
     return;
   }
-  let isThisAuthOrder = parseInt(orderData.user_id) > 0
-  const userId = orderData.user_id
-
-  let apiConfig = config.magento2.api
-  if (orderData.store_code) {
-    if (config.availableStores.indexOf(orderData.store_code) >= 0) {
-      apiConfig = Object.assign({}, apiConfig, { url: apiConfig.url + '/' + orderData.store_code })
-      console.log('> Store code', orderData.store_code)
-    } else {
-      logger.error('Invalid store code', orderData.store_code)
-    }
-  }
-  const api = Magento2Client(apiConfig);
 
-  logger.info('> Order Id', orderData.order_id)
-  logger.info('> Is order authorized?', isThisAuthOrder)
-  logger.info('> User Id', userId)
-
-  let cartId = orderData.cart_id
-  const cartIdPrepare = isThisAuthOrder ? api.cart.create(null, userId) : (cartId ? new Promise((resolve, reject) => {
-    resolve(cartId)
-  }) : api.cart.create(null))
+  var {
```

Comment:
But why?:) what is your argument?
I recently listen talk from Kyle Simpson and I think he made a pretty good argument that `var` is not dead.
`var` is function scope so I think it's okay to use it here.

So what's is your take on `var`?
I also found this short article: https://wesbos.com/is-var-dead .. and I have to agree with these:
> I'm just going to go over two of the leading opinions here. These are both done by some very, very smart people in the JavaScript scene, so I'll let you pick your own.

So I am really curious what is your take on it. :)


Second I have to use `var` for `cartId` because it's re-assigned later.

Reply:
@afirlejczyk `var` is ok in that case, but using only `let` and `const` increase readability. You can use `var` here and it will work in same way as you would use `let`, but with `let` you keep consistence in code. imo it's better to use something that has less power if it's possible and you can use `let` for `cartId`.